jobs: `TestJobInTxn/normal_success` fails when stressing pkg/jobs

Upon running: `make stress PKG=./pkg/jobs TESTFLAGS='-v -logtostderr NONE' TESTFLAGS='-count 500'`, the test `TestJobInTxn/normal_success` fails almost immediately (I believe on the first run) and consistently on master with:
```
=== RUN   TestJobInTxn/normal_success
I200219 18:40:20.174106 40427 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:53450 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-beta.1-187-gbd6dd4122c StartedAt:1582137620083038000 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:53451} ClusterID:e9bc7a1e-8cc0-4664-8912-f69320b7f1ca StartedAt:1582137620083038000 LastUp:1582137620083038000}
I200219 18:40:20.174328 40425 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x67b1440]

goroutine 40690 [running]:
testing.tRunner.func1(0xc00126d200)
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:874 +0x3a3
panic(0x7067fe0, 0x9b14f40)
	/usr/local/Cellar/go/1.13.4/libexec/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).ID(...)
	/Users/pbardea/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:130
github.com/cockroachdb/cockroach/pkg/jobs_test.TestJobInTxn.func6(0xc00126d200)
	/Users/pbardea/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs_test.go:1766 +0x220
testing.tRunner(0xc00126d200, 0xc0032e3dc0)
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:960 +0x350
```

Note, that the test does not flake when stressing the test by itself: `make stress PKG=./pkg/jobs TESTS=TestJobInTxn`

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -1691,6 +1691,8 @@ func TestJobInTxn(t *testing.T) {
 	// Accessed atomically.
 	var hasRun int32
 	var job *jobs.Job
+
+	defer sql.ClearPlanHooks()
 	// Piggy back on BACKUP to be able to create a succeeding test job.
 	sql.AddPlanHook(
 		func(_ context.Context, stmt tree.Statement, phs sql.PlanHookState,
@@ -1756,7 +1758,7 @@ func TestJobInTxn(t *testing.T) {
 		}
 	})
 
-	t.Run("normal success", func(t *testing.T) {
+	t.Run("rollback txn", func(t *testing.T) {
 		txn, err := sqlDB.Begin()
 		require.NoError(t, err)
 		_, err = txn.Exec("BACKUP tobeaborted TO doesnotmattter")
@@ -1767,21 +1769,23 @@ func TestJobInTxn(t *testing.T) {
 		registry := s.JobRegistry().(*jobs.Registry)
 		_, err = registry.LoadJob(ctx, *job.ID())
 		require.Error(t, err, "the job should not exist after the txn is rolled back")
-
 		sqlRunner := sqlutils.MakeSQLRunner(sqlDB)
 		// Just in case the job was scheduled let's wait for it to finish
 		// to avoid a race.
 		sqlRunner.Exec(t, "SHOW JOB WHEN COMPLETE $1", *job.ID())
 		require.Equal(t, int32(0), atomic.LoadInt32(&hasRun),
 			"job has run in transaction before txn commit")
+	})
 
+	t.Run("normal success", func(t *testing.T) {
 		// Now let's actually commit the transaction and check that the job ran.
-		txn, err = sqlDB.Begin()
+		txn, err := sqlDB.Begin()
 		require.NoError(t, err)
 		_, err = txn.Exec("BACKUP tocommit TO foo")
 		require.NoError(t, err)
 		// Committing will block and wait for all jobs to run.
 		require.NoError(t, txn.Commit())
+		registry := s.JobRegistry().(*jobs.Registry)
 		j, err := registry.LoadJob(ctx, *job.ID())
 		require.NoError(t, err, "queued job not found")
 		require.NotEqual(t, int32(0), atomic.LoadInt32(&hasRun),
@@ -1806,5 +1810,4 @@ func TestJobInTxn(t *testing.T) {
 		// failure.
 		require.Error(t, txn.Commit())
 	})
-
 }

--- a/pkg/sql/planhook.go
+++ b/pkg/sql/planhook.go
@@ -111,6 +111,12 @@ func AddPlanHook(f planHookFn) {
 	planHooks = append(planHooks, f)
 }
 
+// ClearPlanHooks is used by tests to clear out any mocked out plan hooks that
+// were registered.
+func ClearPlanHooks() {
+	planHooks = nil
+}
+
 // AddWrappedPlanHook adds a hook used to short-circuit creating a planNode from a
 // tree.Statement. If the returned plan is non-nil, it is used directly by the planner.
 //

